### PR TITLE
Improve user experience with the module

### PR DESCRIPTION
**Pull request by shyim**

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -24,4 +24,9 @@
         <changes lang="en">Fixed meta tags in plugin.xml</changes>
         <changes lang="de">Meta Tags in plugin.xml angepasst</changes>
     </changelog>
+
+    <changelog version="1.0.2">
```
Comment (mitelg) [inline plugin.xml]: we sort from newest version to oldest in all our other plugins :wink:

File: Resources/snippets/backend/swag_tax/main.ini [hunk 2]
```diff
@@ -22,6 +22,9 @@ fourth_card/execute/btn = "Execute now (batch mode)"
 fourth_card/schedule/title = "Schedule"
 fourth_card/text = "Finally, determine whether the changes should be applied immediately or only take effect at a later point.<br><br>Select a date and time for the planned execution. It is also important that the Cron plug-in is active and configured.<br><br>When the changes are executed, the http cache is also cleared."
 
+fourth_card/confirm/title = "Change taxes"
+fourth_card/confirm/message = "Do you really want now change the tax rates?"
```
Comment (PaddyS) [inline Resources/snippets/backend/swag_tax/main.ini]:
```suggestion
fourth_card/confirm/message = "Do you really want to change the tax rates now?"
```

File: Resources/snippets/backend/swag_tax/main.ini [hunk 3]
```diff
@@ -67,3 +76,7 @@ tax_mapping/column/id = "ID"
 tax_mapping/column/name = "Ausgewählte Steuern"
 tax_mapping/column/tax = "Neuer Steuerwert"
 tax_mapping/text = "Wählen Sie im Dropdown-Menü aus, welche der vorhandenen Steuersätze angepasst werden sollen. Nachdem Sie die gewünschten Steuersätze hinzugefügt haben, können Sie in der Spalte \"Neuer Steuerwert\" den neuen Steuersatz eintragen.<br><br>Es werden hierbei nicht die bisherigen Steuersätze überschrieben, sondern neue Einträge erstellt. Zusätzlich werden die ggfs. vorhandenen Steuerregeln übernommen. Sollten in den Regeln also abweichende Werte vorhanden sein, prüfen Sie diese bitte im Anschluss in den Grundeinstellungen."
+
+still_used/product = "Dieser Steuersatz wird noch in Produkten verwendet. Bitte die Seite neuladen."
```
Comment (PaddyS) [inline Resources/snippets/backend/swag_tax/main.ini:80]: "Bitte die Seite neuladen" is not used in the english translation

File: Subscriber/BasicSettingsTaxSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        $snippet = $args->getSubject()->get('snippets')->getNamespace('backend/swag_tax/main');
+
+        $productUsed = $connection->fetchColumn('SELECT 1 FROM s_articles WHERE taxID = ?', [$entityId]);
+        $dispatchUsed = $connection->fetchColumn('SELECT 1 FROM s_premium_dispatch WHERE tax_calculation = ?', [$entityId]);
+        $voucherUsed = $connection->fetchColumn('SELECT 1 FROM s_emarketing_vouchers WHERE taxconfig = ?', [$entityId]);
+
+        if ($productUsed !== false) {
+            throw new \RuntimeException($snippet->get('still_used/product'));
```
Comment (PaddyS) [inline Subscriber/BasicSettingsTaxSubscriber.php]: Wouldn't it be better to collect all messages first?

File: plugin.xml [hunk 5]
```diff
@@ -15,13 +15,19 @@
     <description lang="de">Befristete MwSt.-Senkung vom 01. Juli bis zum 31. Dezember wird die MwSt. in Deutschland von 19 auf 16 sowie von 7 auf 5 % gesenkt.</description>
 
     <compatibility minVersion="5.2.0"/>
-    <changelog version="1.0.0">
```
Comment (PaddyS) [inline plugin.xml:18]: Whoops 🙈

File: Resources/snippets/backend/swag_tax/main.ini (hunk 2 above)
Comment (PaddyS) [reply Resources/snippets/backend/swag_tax/main.ini]: That's still not in 🙈

File: Subscriber/BasicSettingsTaxSubscriber.php (hunk 4 above)
Comment (PaddyS) [reply Subscriber/BasicSettingsTaxSubscriber.php]: I didn't mean it that way. More like: Check for all errors first, and then throw an exception

**Pull request merged**